### ISSUE TITLE
Setting my_lambda and nlambda on MADMMplasso()

### ISSUE THREAD

**Issue by wleoncio** [ocbe-uio/MADMMplasso#54]
Quick question: when calling MADMMplasso(), does it make senst to set `nlambda != ncol(y)`? There doesn't seem to be anything in the code that prevents this, but this bit of code implies that if `my_lambda = NULL`, `lambda_i` is created with length `ncol(y)`:

https://github.com/ocbe-uio/MADMMplasso/blob/081383c17b62b4e51a5f1c9b7736ab28b3d4dccc/R/MADMMplasso.R#L97-L113

This then breaks this later bit of the code, since the vectors `lam[, i]` will have a different size than the vectors `lambda_i[[i]]`:

https://github.com/ocbe-uio/MADMMplasso/blob/081383c17b62b4e51a5f1c9b7736ab28b3d4dccc/R/MADMMplasso.R#L173-L177

**Comment by wleoncio** [comment]
OK, so that means one can't freely assign `nlambda` to whatever integer they want, right? Isn't `nlambda` determined by the number of rows on `my_lambda` (whether passed to `MADMMplasso()` or calculated inside it)?

**Comment by Theo-qua** [comment]
nlambda can be any positive integer so it can be freely specified. I was only saying that it not the same as ncol(y).

**Comment by wleoncio** [comment]
OK, so why does [this code](https://gist.github.com/wleoncio/cc4684bfee99c00fd46db4149be8a05d) give an error (main branch)?

**Comment by wleoncio** [comment]
The code above gives an error of `Error in gg[hh, ] : subscript out of bounds`, implying a connection between `gg` and `nlambda` (through `hh`), but according to the documentation `gg` must be a 2x2 matrix, is that so?

**Comment by Theo-qua** [comment]
Yes, it gave error because of maxgrid = 1L, nlambda = 2L. According to the documentation, nlambda should be less than or equal to maxgrid.

**Comment by Theo-qua** [comment]
gg is regenerated in the function with a length equal to maxgrid before the loop.

**Comment by wleoncio** [comment]
Great, thanks! I read that part, but didn't catch the exception. We should validate that `nlambda <= maxgrid` to avoid unclear error messages (and unnecessary computation), agreed? I can open a PR with a quick fix.

Regarding `gg`, I'm confused now. I can't find it being created inside `MADMMplasso()`, only used to calculate `lammax`, `gg1`, `gg2` and `gg3` (see [here](https://github.com/ocbe-uio/MADMMplasso/blob/081383c17b62b4e51a5f1c9b7736ab28b3d4dccc/R/MADMMplasso.R#L110-L129)). `gg` even seems like a mandatory argument, as not passing it (or passing `NULL`, which is the default) results in error.

**Comment by Theo-qua** [comment]
I agree with you. I reassigned gg to gg3 in the MADMMplasso() after calculating gg3. I did not consider to cause any confusion since this is internal assignment. So gg is changed from the initial gg in the call to gg3.
